### DISCUSSION TITLE
Ambient noise correlation

### DISCUSSION THREAD

**Discussion by ariellellouch** [DASDAE/dascore#276] [Ideas]
Hi all,

I was thinking it would be neat to have some basic functionality for ambient noise cross-correlation. We probably need to define window intervals, normalization method, etc., but the infrastructure is there

**Comment by ahmadtourei** [comment]
Hi @ariellellouch 
Great idea! However, I'm unsure if we should plan to include it in DASCore or in a separate application package later. I think it would be great to have some basic functions for CC here and include more detailed codes and functions (normalization, whitening, etc) in the "Near-surface imaging" package later. @d-chambers What do you think?
I'd be happy to work with you on implementing ambient noise interferometry functions!

**Reply by ariellellouch** [reply]
Sorry, I am not really aware of the detailed planning, so some of my suggestions might be redundant :)
I was just looking briefly at the projects and existing code, and it struck me as a nice to-have feature.

The only comment I would make is that ambient noise on long fibers could potentially go into the km scale in imaging, so the package might be "lost" if part of near-surface imaging only

**Comment by d-chambers** [comment]
I think having some of the basic correlation machinery in DASCore is a great idea, but we should leave optimization which require heavy dependencies (eg MPI/GPU integration) to specialized libraries like noisepy. 

It would be nice to have separate patch methods for each step of the process. For example, one method for spectral whitening, another for 1 bit normalization, etc. I know there are lots of permutations for the correlation workflow, but we don't need to implement all of them.

Thoughts?

**Comment by ahmadtourei** [comment]
I agree. I think it makes sense to start including some of the common preprocessing algorithms and cross-correlation calculations. 

@ariellellouch Please feel free to start working on these if you got a chance. Let me know if you want to discuss planning/coding. I have already created some scripts for DAS ambient noise interferometry a while ago that may be helpful. Not specifically for DASCore though.

**Comment by ariellellouch** [comment]
@ahmadtourei I haven't done it before myself, so you're ahead of me - just getting into the topic through one of my students

I can probably help with whitening, normalizations, etc., like @d-chambers suggested, but with the current situation in Israel it will take me longer than usual

**Comment by d-chambers** [comment]
I think we have most the items discussed here now, (whitening, normalization, correlation) so I will close for now. Feel free to re-open if we are missing something (or just make a PR with the new features 😉)
